nexus/reconfigurator/planning/src/mgs_updates/mod.rs

```diff
@@ -996,14 +1363,15 @@ mod test {
         assert_eq!(old_update.slot_id, new_update.slot_id);
         assert_eq!(old_update.artifact_hash, new_update.artifact_hash);
         assert_eq!(old_update.artifact_version, new_update.artifact_version);
-        let PendingMgsUpdateDetails::Sp {
-            expected_active_version: new_expected_active_version,
+        let PendingMgsUpdateDetails::Rot {
+            expected_active_slot: new_expected_active_slot,
```

Comment:
Same here -- I guess you changed a bunch of tests in a way that either of two updates were possible and that's causing it to pick the RoT one first.  I would force the test to pick one or the other just to make it clearer to the reader (and less brittle if we did somehow change the order of these, but I'm less worried about that).